File: src/main/java/org/example/model/Country.java
package org.example.model;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;

@Entity
public class Country {

    @Id
    private String code;

    @Column(nullable = false)
    private String code2;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String continent;

    @Column(nullable = false)
    private String region;

    @Column(nullable = false)
    private Long population;

    @Column(name = "life_expectancy")
    private Double lifeExpectancy;

    @Column(name = "surface_area", nullable = false)
    private Double surfaceArea;

    @Column(name = "government_form", nullable = false)
    private String governmentForm;

    @Column(name = "local_name", nullable = false)
    private String localName;

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getLocalName() {
        return localName;
    }

    public void setLocalName(String localName) {
        this.localName = localName;
    }

    public String getCode2() {
        return code2;
    }

    public void setCode2(String code2) {
        this.code2 = code2;
    }

    public String getContinent() {
        return continent;
    }

    public void setContinent(String continent) {
        this.continent = continent;
    }

    public String getRegion() {
        return region;
    }

    public void setRegion(String region) {
        this.region = region;
    }

    public Double getSurfaceArea() {
        return surfaceArea;
    }

    public void setSurfaceArea(Double surfaceArea) {
        this.surfaceArea = surfaceArea;
    }

    public String getGovernmentForm() {
        return governmentForm;
    }

    public void setGovernmentForm(String governmentForm) {
        this.governmentForm = governmentForm;
    }

    public Long getPopulation() {
        return population;
    }

    public void setPopulation(Long population) {
        this.population = population;
    }

    public Double getLifeExpectancy() {
        return lifeExpectancy;
    }

    public void setLifeExpectancy(Double lifeExpectancy) {
        this.lifeExpectancy = lifeExpectancy;
    }
}

File: src/main/kotlin/org/example/model/Country.kt
package org.example.model

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

/**
 * As opposed to what comes naturally here, using data classes as JPA entities is generally discouraged.
 *
 * Refs:
 * - https://www.baeldung.com/kotlin/jpa
 * - https://www.baeldung.com/kotlin/jpa#data-classes
 */
@Entity
@Table(name = "country")
open class Country(
    @Id
    open val code: String,

    @Column(nullable = false)
    open val code2: String,

    @Column(nullable = false)
    open val name: String,

    @Column(nullable = false)
    open val continent: String,

    @Column(nullable = false)
    open val region: String,

    @Column(nullable = false)
    open val population: Long,

    @Column(name = "life_expectancy")
    open val lifeExpectancy: Double? = null,

    @Column(name = "surface_area", nullable = false)
    open val surfaceArea: Double,

    @Column(name = "government_form", nullable = false)
    open val governmentForm: String,

    @Column(name = "local_name", nullable = false)
    open val localName: String
)


